scripts/src/owners/owners_file.py

```diff
@@ -8,70 +9,67 @@
     from yaml import SafeLoader
 
 
+class OwnersFileError(Exception):
+    pass
+
+
 def get_owner_data(category, organization, chart):
     path = os.path.join("charts", category, organization, chart, "OWNERS")
-    status, owner_content = get_owner_data_from_file(path)
-    return status, owner_content
+    success = True
+
+    try:
+        owner_content = get_owner_data_from_file(path)
+    except OwnersFileError as e:
+        print(f"Error getting OWNERS file data: {e}")
+        success = False
+
+    return success, owner_content
 
 
 def get_owner_data_from_file(owner_path):
     try:
         with open(owner_path) as owner_data:
             owner_content = yaml.load(owner_data, Loader=SafeLoader)
-        return True, owner_content
-    except Exception as err:
-        print(f"Exception loading OWNERS file: {err}")
-        return False, ""
+    except yaml.YAMLError as e:
+        print(f"Exception loading OWNERS file: {e}")
+        raise OwnersFileError from e
+    except OSError as e:
+        print(f"Error opening OWNERS file: {e}")
+        raise OwnersFileError from e
+
+    return owner_content
 
 
 def get_vendor(owner_data):
-    vendor = ""
-    try:
-        vendor = owner_data["vendor"]["name"]
-    except Exception:
-        pass
-    return vendor
+    vendor_name = ""
+    with contextlib.suppress(KeyError):
+        vendor_name = owner_data["vendor"]["name"]
+    return vendor_name
```

Comment:
I was originally going to suggest this instead (to avoid importing `contextlib`):

```python
return owner_data.get("vendor", {}).get("name", "")
```

But, actually, my implementation (and yours) both suffer from a failure if owner_data is not a dictionary. AttributeError would get thrown for mine (calling the get method), and TypeError for yours (NoneType is not subscriptable).

Better to leave this as is?

---

[EDIT]: I'm seeing now that there are several places where this logic was being simplified, but we'll run into the problem cases because we can't assert the owner_data input's type.